### PR TITLE
Comply with requirements for MSFT-owned nuget packages

### PR DESCRIPTION
**Pull request by swannman**
- Update nuget IDs to use reserved Microsoft.O365.Security prefix
- Update assembly namespace to match updated nuget id
- Use Microsoft ESRP strong name and authenticode signature
- Disable standards conformance for tests project to work around compiler bug ([ref](https://developercommunity.visualstudio.com/content/problem/288560/cppunittestassert-doesnt-compile-with-permissive-i.html))

### CODE REVIEW

File: O365.Security.Native.ETW.Debug.nuspec [hunk 1]
```diff
@@ -1,24 +1,24 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
     <metadata>
-        <id>O365.Security.Native.ETW.Debug</id>
+        <id>Microsoft.O365.Security.Native.ETW.Debug</id>
         <version>1.0.13</version>
```
Comment (zacbrown) [inline O365.Security.Native.ETW.Debug.nuspec]: We should bump this to 2.0. By changing the root namespace, it's a breaking change for users.

File: tests/krabstests/krabstests.vcxproj [hunk 2]
```diff
@@ -98,7 +98,7 @@
       <AdditionalIncludeDirectories>$(VCInstallDir)UnitTest\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>_DEBUG;TYPEASSERT;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <UseFullPaths>true</UseFullPaths>
-      <ConformanceMode>true</ConformanceMode>
```
Comment (zacbrown) [inline tests/krabstests/krabstests.vcxproj]: Why are we disable conformance mode?

File: O365.Security.Native.ETW.nuspec [hunk 3]
```diff
@@ -1,24 +1,24 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
     <metadata>
-        <id>O365.Security.Native.ETW</id>
+        <id>Microsoft.O365.Security.Native.ETW</id>
         <version>1.0.13</version>
```
Comment (zacbrown) [inline O365.Security.Native.ETW.nuspec]: 2.0

File: tests/krabstests/krabstests.vcxproj (hunk 2 above)
Comment (swannman) [reply tests/krabstests/krabstests.vcxproj]: It breaks the VSTS build, and we haven't been able to find a workaround: https://developercommunity.visualstudio.com/content/problem/288560/cppunittestassert-doesnt-compile-with-permissive-i.html

File: O365.Security.Native.ETW.nuspec (hunk 3 above)
Comment (swannman) [reply O365.Security.Native.ETW.nuspec]: Fixed

File: O365.Security.Native.ETW.Debug.nuspec (hunk 1 above)
Comment (swannman) [reply O365.Security.Native.ETW.Debug.nuspec]: Fixed

File: tests/krabstests/krabstests.vcxproj (hunk 2 above)
Comment (zacbrown) [reply tests/krabstests/krabstests.vcxproj]: Ah nope - I missed your bullet point above. That's fine.

Comment (swannman) [reply tests/krabstests/krabstests.vcxproj]: thanks!

**Pull request merged**